add-ons/istio/README.adoc

```diff
@@ -0,0 +1,84 @@
+:linkattrs:
+
+= Istio Add-on
+
+This add-on deploys https://Istio.io/[Istio] service mesh on to Minishift-provisioned OpenShift cluster.
+
+== Minishift profile setup
+
+The standard Istio deployment creates few pods and its related resources. 
+
+Considering this, it is ideal to setup a minishift profile with the following configuration:
+
+[code,sh]
+----
+$ minishift profile set servicemesh <1>
+$ minishift config set memory 8GB <2>
+$ minishift config set cpus 3 <3>
+$ minishift config set openshift-version v3.7.0 <4>
+$ minishift addon enable admin-user <5>
+$ minishift addon enable anyuid <6>
+$ minishift start <7>
+----
+
+<1> Create a new Minishift profile named **servicemesh**
+<2> Use 8GB of memory for istio and apps that you might deploy later
+<3> Since Istio minimum deploys approximately __9__ pods, 3 cpus will give a max of 30 pods
+<4> Istio 0.2.x and above requires OpenShift version 3.7.x or above
+<5> Enable Minishift `admin-user` add-on 
+<6> Enable Minishift `anyuid` add-on .  For more details on OpenShift scc refer https://docs.openshift.org/latest/admin_guide/manage_scc.html[Managing Security Context Constraints]
+<7> Since we have set the profile to be **servicemesh**, this command starts Minishift with profile `servicemesh`
+
+== Preparing Istio 
+Download Istio corresponding to your platform from https://github.com/istio/istio/releases[Download an Istio release] extract it to a folder of your choice
+and export that folder as `ISTIO_HOME`
+
+On macOS/Linux machines you can do like:
+
+[code,sh]
+----
+$ curl -L https://git.io/getLatestIstio | sh -
+$ ISTIO_HOME=$(find `pwd` -name \*istio\* -type d -maxdepth 1 -print -quit)
```

Comment:
`find` will cause a warning to be printed with `-maxdepth` where it currently is. I also don't think all of the flags given are necessary, but I'm not sure. In either case, `-maxdepth` should be placed before `-name`, and this command should work as well as what's currently documented (without creating another subshell):

`ISTIO_HOME=$(find -maxdepth 1 -name \*istio\* -type d)`

Reply:
this does not work on macOS. `ISTIO_HOME=$(find `pwd` -maxdepth 1 -name \*istio\* -type d)` this is the one that works on both macOS and linux so updating to this